docs/debug/mpcie1_mpcie2_ethernet_card.md

```diff
@@ -0,0 +1,124 @@
+Problems with APU2 when using Mini PCI-Express Dual Gigabit Ethernet Controller Card
+====================================================================================
+
+## Problems
+
+1. When Ethernet Controller Card is placed in mPCIe1 slot (J14) placed extension cards 
+is recognized by OS but standard Ethernet output placed on APU2 board can't receive 
+correct IP adress. So it doesn't work correct. 
+
+2. When Ethernet Controller Card is placed in mPCIe2 slot (J13) 2 additional Ethernet 
+controller are not visible in OS. Standard Ethernet output placed on APU2 board 
+works well.
+
+> Device detection has been checked using `ip a`, `lspci` and `dmesg`. The results are 
+shown in the table below.
+
+```
+	+-----+-----------------------------+
+	| dev |          visibility         |
+	+-----+--------------+--------------+
+	| eth | mPCIe1 (J14) | mPCIe2 (J13) |
+	+-----+--------------+--------------+
+	|  0  |     yes      |     yes      |
+	|  1  |     yes      |     yes      |
+	|  2  |     yes      |     yes      |
+	|  3  |     yes      |     no       |
+	|  4  |     yes      |     no       |
+	+--------------------+--------------+
+```
+
+## Mini PCI-Express Dual Gigabit Ethernet Controller Card elements
+
+On a board are located PCI bridge [ASMedia Technology Inc. Device 1182](http://www.asmedia.com.tw/eng/e_show_products.php?cate_index=112&item=134) and 2 Ethernet 
+controllers [Realtek Semiconductor Co., Ltd. RTL8111/8168/8411](http://www.realtek.com/products/productsView.aspx?Langid=1&PNid=13&PFid=5&Level=5&Conn=4).
+
+## Booting lags
+
+When Ethernet Controller is placed in the mPCIe1 slot (J14) and booting is at network 
+devices setup time occurs long lag. In this case `eth0` device can not get correct IP. 
+After this next booting processes take same normal time. The reason may be multiple 
+failed attempts to obtain an IP address.
+That problem doesn't exist for the mPCIe2 slot (J13).
+
+## Boot logs comparison
+
+Level of the detail of booting logs was set at `6:DEBUG`.
+
+> The full list varies between logs is placed at the end file.
+
+The most suspicious difference in logs is placed below.
+
+For `mPCIe2`:
+
+```	
+	PCI: Static device PCI: 00:02.5 not found, disabling it.
+```
+For `mPCIe1`:
+
+```
+	PCI: 00:02.5 subordinate bus PCI Express
+	PCI: 00:02.5 [1022/1439] enabled
+```
+
+This suggests a problem with device recognition when Ethernet Controller is 
+placed in the mPCIe1 slot (J14). The reason for this can be rivalry of devices
+or incorrect designed interrupt system.
```

Comment:
@acihy I'm not sure if this statement not going too far. We are not sure about that and have no evidence. Speaking about irq I would say it can be incorrectly configured, but designed ?

`rivalry` word sounds weird in this context and probable translated directly. I would say we may think about `race condition` or `concurrent accees`